src/app/Main/Body/QueryPage/Drawer/Drawer-Layout.component.tsx

```diff
@@ -98,16 +106,24 @@ export function DrawerLayoutComponent(props: DrawerLayoutProps): JSX.Element {
             <div className={classes.drawerContainer}>
                 <DrawerFilterComponent
                     dataUniqueValues={props.dataUniqueValues}
-                    selectedFilter={props.selectedFilter}
-                    mainFilterAttributes={props.mainFilterAttributes}
+                    selectedFilter={props.filterInfo.selectedFilter}
+                    filterToDisplay={props.filterInfo.displayedFilter}
+                    mainFilterAttributes={props.filterInfo.mainFilter}
                     onFilterChange={handleChangeFilter}
                     onFilterRemoveAll={handleRemoveAllFilter}
                 />
-                <Divider variant="middle" css={dividerStyle} />
+                <FilterSettingDialogComponent
```

Comment:
Consider moving the FilterDialog to the DrawerFilterComponent, they describe the same context and should stick together.

Reply:
Done. Commit "Move dialog & make dialog stupid"